packages/graphql-server/src/functions/graphql.ts

```diff
@@ -209,120 +194,61 @@ export const createGraphQLHandler = ({
     },
   })
 
-  function buildRequestObject(event: APIGatewayProxyEvent) {
-    const requestHeaders = new Headers()
-    const supportsMultiValueHeaders =
-      event.multiValueHeaders && Object.keys(event.multiValueHeaders).length > 0
-    // Avoid duplicating header values, because Yoga gets confused with CORS
-    if (supportsMultiValueHeaders) {
-      for (const headerName in event.multiValueHeaders) {
-        const headerValues = event.multiValueHeaders[headerName]
-        if (headerValues) {
-          for (const headerValue of headerValues) {
-            requestHeaders.append(headerName, headerValue)
-          }
-        }
-      }
-    } else {
-      for (const headerName in event.headers) {
-        const headerValue = event.headers[headerName]
-        if (headerValue) {
-          requestHeaders.append(headerName, headerValue)
-        }
-      }
-    }
-
-    const protocol = isDevEnv ? 'http' : 'https'
-
-    const requestUrl = new URL(
-      event.path,
-      protocol + '://' + (event.requestContext?.domainName || 'localhost')
-    )
-
-    if (event.multiValueQueryStringParameters) {
```

Comment:
Note: Need to confirm multi-value header tests pass. Or did this logic get rolled into v3?

Reply:
I think having both `multi` and singular ones are unnecesary because singular has multiple values with `,` so that's ok. Fetch API already catches the difference but we can test it in a real env for sure.